src/model/CoilCoolingDXCurveFitPerformance.cpp

```diff
@@ -99,13 +99,13 @@ namespace model {
       std::vector<ModelObject> result;
 
       // These are ResourceObjects, so they shouldn't really be children except for OS App / IG
-      result.push_back(baseOperatingMode());
```

Comment:
I rebased the change(s) onto my codebase and rebuild and cause the failure of serveral tests.

```
[  FAILED  ] 3 tests, listed below:
[  FAILED  ] ModelFixture.CoilCoolingDX_clone
[  FAILED  ] ModelFixture.CoilCoolingDXCurveFitPerformance_clone
[  FAILED  ] ModelFixture.CoilCoolingDXCurveFitOperatingMode_clone
```

Do you mind review the tests?

Reply:
Yeah, I'm still not sure how to fix it properly. I need to revisit this at some point.

Reply:
I deep dived into the unittests and find the issue is happening in 


```
  EXPECT_EQ(0u, model.getConcreteModelObjects<CoilCoolingDXCurveFitOperatingMode>().size());
```

Which the `model.getConcreteModelObjects<CoilCoolingDXCurveFitOperatingMode>().size()`'s value is 1.

Reply:
I have access to Jenkins FYI, so I have access to the failed runs to see the failures.

The solution to having both clone to same model and clone to another is not trivial though. I'll eventually get back to this PR, it's a minor bug with a rather *expensive* solution: one that requires reimplementing or overriding the internals of ResourceObject and ParentObject (getRecursiveResources / getRecursiveChildren) and how things are added at Workspace level (addObjects). That portion of the sdk is complicated and finicky.
I've converted the PR to draft in the meantime for clarity.

Reply:
If someone is burning to fix this though please go ahead. I think I have a jupyter notebook somewhere where I've rewritten the getRecursiveXXX stuff to be able to test changes, I could try to locate it.